sprints/dashboard/tasks.py

```diff
@@ -413,3 +434,174 @@ def unflag_tickets_task() -> None:
+
+
+@celery_app.task(ignore_result=True)
+def create_estimation_session_task() -> None:
+    """
+    Create a new empty estimation session for every cell.
+
+    An empty estimation session does not contain any issues or participants. The current user is added as the scrum
+    master of the session, as without this permission it wouldn't be possible to make any future modifications.
+
+    WARNING
+    Creating a session without issues causes some chaos in Jira, as the `/session/async/{sessionId}/rounds/` endpoint
+    returns HTTP 500 in such case. It does not break other API calls, so operations like updating, closing, and deleting
+    the session (via the API) work correctly. It makes the session unusable via the browser by breaking two views:
+        - estimation,
+        - configuration.
+    Therefore, the decision is to avoid adding the participants to the session until there are issues that can be added
+    too. Assuming that the sessions are fully automated, and don't require any manual interventions in the beginning,
+    this should not cause any troubles.
+    """
+    with connect_to_jira() as conn:
+        session_name = get_next_poker_session_name(conn)
+        for cell in get_cells(conn):
+            if not settings.DEBUG:  # We really don't want to trigger this in the dev environment.
+                conn.create_poker_session(
+                    board_id=cell.board_id,
+                    name=session_name,
+                    issues=[],
+                    participants=[],
+                    scrum_masters=[conn.myself()['key']],
+                    send_invitations=False,
+                )
+
+
+@celery_app.task(ignore_result=True)
+def update_estimation_session_task() -> None:
+    """
+    Update estimation session's issues and participants.
+
+    If no issues exist for the session, then it will not be updated. The reasoning behind this has been described in the
+    `create_estimation_session_task` function.
+
+    This does not override the manual additions to the session - i.e. if an issue or user has been added manually to the
+    session, then it will be retained, as it merges available issues and participants with the applied ones.
+    However, any removed items (e.g. an issue scheduled for the next sprint, or a user who is a member of the cell)
+    will be added back automatically.
+
+    FIXME: This adds all cell members as scrum masters to the session, because at the moment we do not have a way to
+           determine whether the member is a part of the core team. We can restrict this in the future, if needed.
+    """
+    with connect_to_jira() as conn:
+        session_name = get_next_poker_session_name(conn)
+        issues = get_unestimated_next_sprint_issues(conn)
+        # FIXME: This will not work for more than 1000 users. To support it, add `startAt` to handle the pagination.
+        all_users = conn.search_users("''", maxResults=1000)  # Searching for the "quotes" returns all users.
```

Comment:
Yay, weird API design, which is `Gathering Interest` [since 2012](https://jira.atlassian.com/browse/JRASERVER-29069?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&showAll=true).